File: LifecycleExample/LifecycleExampleJava/app/src/main/java/com/tinlone/lifecycleexamplejava/ProcessLifecycleObserver.java
package com.tinlone.lifecycleexamplejava;

import android.util.Log;

import androidx.lifecycle.Lifecycle;
import androidx.lifecycle.LifecycleObserver;
import androidx.lifecycle.OnLifecycleEvent;

public class ProcessLifecycleObserver implements LifecycleObserver {

    private boolean isForeground = false;

    @OnLifecycleEvent(Lifecycle.Event.ON_START)
    public void onForeground() {
        Log.i("loglog", "onForeground: ON_START");
        isForeground = true;
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_STOP)
    public void onBackground() {
        Log.i("loglog", "onBackground: ON_STOP");
        isForeground = false;
    }

    public boolean isForeground() {
        return isForeground;
    }

}

File: LifecycleExample/LifecycleExampleKotlin/app/src/main/java/com/tinlone/lifecycleexamplekotlin/ProcessLifecycleObserver.kt
package com.tinlone.lifecycleexamplekotlin

import android.util.Log
import androidx.lifecycle.Lifecycle
import androidx.lifecycle.LifecycleObserver
import androidx.lifecycle.OnLifecycleEvent

class ProcessLifecycleObserver : LifecycleObserver {
    private var isForeground = false

    @OnLifecycleEvent(Lifecycle.Event.ON_START)
    fun onForeground() {
        Log.i("loglog", "onForeground: ON_START")
        isForeground = true
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_STOP)
    fun onBackground() {
        Log.i("loglog", "onBackground: ON_STOP")
        isForeground = false
    }

    fun isForeground(): Boolean {
        return isForeground
    }
}